client/extensions/woocommerce/state/action-list/actions.js

```diff
@@ -0,0 +1,66 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_ACTION_LIST_CREATE,
+	WOOCOMMERCE_ACTION_LIST_CLEAR,
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_END,
+} from '../action-types';
+
+/**
+ * Action Creator: Create new Action List.
+ *
+ * @param {Object} actionList The sequential actionList to be carried out
+ * @return {Object} action
+ */
+export function actionListCreate( actionList ) {
+	return {
+		type: WOOCOMMERCE_ACTION_LIST_CREATE,
+		actionList,
+	};
+}
+
+/**
+ * Action Creator: Clear the current product Action List.
+ * @return {Object} action
+ */
+export function actionListClear() {
+	return {
+		type: WOOCOMMERCE_ACTION_LIST_CLEAR,
+	};
+}
+
+/**
+ * Action Creator: Mark step as started.
+ *
+ * @param {Number} stepIndex The index of the step.
+ * @param {Date} [time=Date.now()] Optional timestamp.
+ * @return {Object} action
+ */
+export function actionListStepStarted( stepIndex, time = Date.now() ) {
```

Comment:
I like the look `actionListStepStart`(to closer match the action type) or `actionListMarkStepStarted`. 

`actionListStepStarted` feels more like something that would happen after a step started, rather than the thing that marks the step as started.